iceoryx2-ffi/cxx/include/iox/slice.hpp

```diff
@@ -16,38 +16,134 @@
 #include "iox/assertions_addendum.hpp"
 
 #include <cstdint>
+#include <type_traits>
 
 namespace iox {
+
 template <typename T>
 class Slice {
   public:
-    using Iterator = T*;
-    using ConstIterator = const T*;
-    using ValueType = T;
-
-    auto size() const -> uint64_t {
-        IOX_TODO();
-    }
-    auto operator[](const uint64_t n) const -> const T& {
-        IOX_TODO();
-    }
-    auto operator[](const uint64_t n) -> T& {
-        IOX_TODO();
-    }
-    auto begin() -> Iterator {
-        IOX_TODO();
-    }
-    auto begin() const -> ConstIterator {
-        IOX_TODO();
-    }
-    auto end() -> Iterator {
-        IOX_TODO();
-    }
-    auto end() const -> ConstIterator {
-        IOX_TODO();
-    }
+    using Iterator = std::conditional_t<std::is_const_v<T>, const T*, T*>;
+    using ValueType = std::remove_const_t<T>;
+
+    Slice(T* data, uint64_t number_of_elements);
+
+    /// @brief Returns the size of the slice in bytes.
+    /// @return The size of the slice in bytes.
+    auto size() const -> uint64_t;
```

Comment:
Food for thought: `size` might be ambiguous. For example `std::vector<T>::size()` return the number of elements. Since there is already a `number_of_elements` method, how about renaming this to `number_of_bytes` or `size_in_bytes` to remove the ambiguity

Reply:
@elBoberido Fixed